lib/src/arithmetic/multiplier.dart

```diff
@@ -227,31 +353,64 @@ class MutiplyOnly extends MultiplyAccumulate {
   @override
   Logic get accumulate => output('accumulate');
 
+  static String _genName(
+          Multiplier Function(Logic a, Logic b,
+                  {Logic? selectSignedMultiplicand,
+                  Logic? selectSignedMultiplier})
+              fn,
+          Logic a,
+          Logic b,
+          Logic? selectSignedMultiplicand,
+          Logic? selectSignedMultiplier) =>
+      fn(a, b,
+              selectSignedMultiplicand: selectSignedMultiplicand,
+              selectSignedMultiplier: selectSignedMultiplier)
+          .name;
+
   /// Construct a MultiplyAccumulate that only multiplies to enable
   /// using the same tester with zero accumulate addend [c].
-  MutiplyOnly(super.a, super.b, super.c,
-      Multiplier Function(Logic a, Logic b, {Logic? selectSigned}) mulGen,
-      {super.signed = false,
-      Logic? selectSigned}) // Will be overrwridden by multiplyGenerator
-      : super(
-            name: 'Multiply Only: '
-                '${mulGen.call(a, b, selectSigned: selectSigned).name}') {
-    final Logic? internalSelectSigned;
-
-    if (selectSigned != null) {
-      internalSelectSigned = addInput('selectSigned', selectSigned);
-    } else {
-      internalSelectSigned = null;
+  MutiplyOnly(
+    super.a,
+    super.b,
+    super.c,
+    Multiplier Function(Logic a, Logic b,
+            {Logic? selectSignedMultiplicand, Logic? selectSignedMultiplier})
+        mulGen, {
+    super.signedMultiplicand = false,
+    super.signedMultiplier = false,
+    super.signedAddend = false,
+    super.selectSignedMultiplicand,
+    super.selectSignedMultiplier,
+    super.selectSignedAddend,
+  }) // Will be overrwridden by multiplyGenerator
+  : super(
+            // ignore: prefer_interpolation_to_compose_strings
+            name: 'Multiply Only: ' +
+                _genName(mulGen, a, b, selectSignedMultiplicand,
```

Comment:
is this constructing a temporary piece of hardware just for the sake of getting its name?

Reply:
Yes.  It is done only at the beginning of each test to give me a clear idea of the parameters passed, which can affect the name.  A bit experimental, but the name of a module could be extended with some of its parameter settings -- very useful in testing, but maybe not so much in instantiation, though perhaps somewhat useful.